src/goto-symex/goto_symex_state.cpp

```diff
@@ -660,22 +660,16 @@ void goto_symex_statet::rename(
     rename(array_type.subtype(), irep_idt(), ns, level);
     rename(array_type.size(), ns, level);
   }
-  else if(type.id()==ID_struct ||
-          type.id()==ID_union ||
-          type.id()==ID_class)
```

Comment:
This is not refactoring as the commit message suggests and this change causes some tests (in particular jbmc/regression/jbmc-strings/StringSubstring) to fail.